### PR TITLE
prepare for lnd 0.16

### PR DESCRIPTION
**Pull request by guggero**
This PR adds compile time compatibility with lnd 0.16.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,19 +15,19 @@ require (
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.5.0
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/lightninglabs/lightning-node-connect/hashmailrpc v1.0.2
-	github.com/lightninglabs/lndclient v0.16.0-6
-	github.com/lightningnetwork/lnd v0.15.0-beta.rc6.0.20221207163254-a0385a535b66
-	github.com/lightningnetwork/lnd/cert v1.1.1
+	github.com/lightninglabs/lndclient v0.16.0-8
+	github.com/lightningnetwork/lnd v0.15.0-beta.rc6.0.20230207162325-a233df99c49d
```
Comment (bhandras) [inline go.mod:19]: With the arrival of the rc1, can quickly fixup deps and merge?

**Review by bhandras — APPROVED** [review]
LGTM 🙏

**Review by positiveblue — APPROVED** [review]
:fire:

File: go.mod [hunk 2]
```diff
@@ -121,35 +127,47 @@ require (
...
+	modernc.org/opt v0.1.3 // indirect
+	modernc.org/sqlite v1.20.3 // indirect
+	modernc.org/strutil v1.1.3 // indirect
+	modernc.org/token v1.0.1 // indirect
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
-// Fix etcd token renewal issue https://github.com/etcd-io/etcd/pull/13262.
-replace go.etcd.io/etcd/client/v3 => github.com/lightninglabs/etcd/client/v3 v3.5.1-retry-patch
+// We need to use grpc v1.39.0 because of a change in how HTTP errors are
```
Comment (positiveblue) [inline go.mod:170]: :pray:

**Pull request merged**